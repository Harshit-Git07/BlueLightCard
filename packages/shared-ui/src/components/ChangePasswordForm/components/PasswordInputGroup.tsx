import { FC } from 'react';
import PasswordInput from '../../PasswordInput';
import { CurrentFormState, PasswordField } from '../types';
import copy from '../copy';

export type PasswordInputGroupProps = {
  formState: CurrentFormState;
  updatePasswordValue: (updatedValue: string, fieldToUpdate: PasswordField) => void;
  onBlur: (field: PasswordField) => void;
};
export const PasswordInputGroup: FC<PasswordInputGroupProps> = ({
  formState,
  updatePasswordValue,
  onBlur,
}) => {
  const fieldIsValid = (field: PasswordField) =>
    !formState[field].value ? undefined : !formState[field].error;
  return (
    <div className="flex flex-col gap-4">
      <PasswordInput
        password={formState.currentPassword.value}
        onChange={(newVal) => updatePasswordValue(newVal, 'currentPassword')}
        onBlur={() => onBlur('currentPassword')}
        isValid={fieldIsValid('currentPassword')}
        label={copy.currentPasswordInput.label}
        placeholderText={copy.currentPasswordInput.placeholder}
        infoMessage={formState.currentPassword.error}
        showIcon={false}
        hideRequirements
      />
      <PasswordInput
        password={formState.newPassword.value}
        onChange={(newVal) => updatePasswordValue(newVal, 'newPassword')}
        onBlur={() => onBlur('newPassword')}
        isValid={fieldIsValid('newPassword')}
        label={copy.newPasswordInput.label}
        placeholderText={copy.newPasswordInput.placeholder}
        infoMessage={formState.newPassword.error}
        showIcon={false}
      />
      <PasswordInput
        password={formState.newPasswordConfirm.value}
        onChange={(newVal) => updatePasswordValue(newVal, 'newPasswordConfirm')}
        onBlur={() => onBlur('newPasswordConfirm')}
        isValid={fieldIsValid('newPasswordConfirm')}
        label={copy.newPasswordConfirmInput.label}
        placeholderText={copy.newPasswordConfirmInput.placeholder}
        infoMessage={formState.newPasswordConfirm.error}
        showIcon={false}
        hideRequirements
      />
    </div>
  );
};
