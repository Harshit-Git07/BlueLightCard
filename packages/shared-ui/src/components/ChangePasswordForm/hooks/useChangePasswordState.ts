import { useState } from 'react';
import { useFormValidation } from './useFormValidation';
import { CurrentFormState, InputElementState, PasswordField } from '../types';
import { useChangePasswordPut } from './useChangePasswordPut';
import useMemberProfileGet from '../../../hooks/useMemberProfileGet';

const initialElementState: InputElementState = {
  value: '',
};

const initialFormState: CurrentFormState = {
  currentPassword: initialElementState,
  newPassword: initialElementState,
  newPasswordConfirm: initialElementState,
};

export const useChangePasswordState = () => {
  const [formState, setFormState] = useState<CurrentFormState>(initialFormState);
  const { memberProfile } = useMemberProfileGet();
  const { validationResult, validateForm } = useFormValidation(formState);
  const { mutateAsync, isPending } = useChangePasswordPut();

  const updateValidation = (updatedForm: CurrentFormState, field: PasswordField) => {
    const validation = validateForm(updatedForm);
    if (!validation.success) {
      const fieldError = validation.error.issues.find((issue) => issue.path.includes(field));
      updatePasswordError(fieldError?.message ?? '', field);
    }
  };

  const updatePasswordValue = (updatedValue: string, fieldToUpdate: PasswordField) => {
    const newState = { ...formState, [fieldToUpdate]: { value: updatedValue } };
    setFormState(newState);
    updateValidation(newState, fieldToUpdate);
  };

  const updatePasswordError = (errorMessage: string, fieldToUpdate: PasswordField) => {
    setFormState((prevState) => ({
      ...prevState,
      [fieldToUpdate]: { ...prevState[fieldToUpdate], error: errorMessage },
    }));
  };

  const makeApiRequest = async () => {
    if (memberProfile) {
      return mutateAsync({
        email: memberProfile.email,
        currentPassword: formState.currentPassword.value,
        newPassword: formState.newPassword.value,
      });
    }
  };

  return {
    formState,
    saveDisabled: !validationResult.success || isPending,
    updatePasswordValue,
    updatePasswordError,
    makeApiRequest,
  };
};

export default useChangePasswordState;
