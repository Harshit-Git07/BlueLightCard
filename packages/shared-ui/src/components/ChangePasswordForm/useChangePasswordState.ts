import { useState } from 'react';
import { useFormValidation } from './useFormValidation';
import { CurrentFormState, InputElementState, PasswordField } from './types';
import { useChangePasswordPut } from './useChangePasswordPut';

const initialElementState: InputElementState = {
  value: '',
};

export const initialFormState: CurrentFormState = {
  currentPassword: initialElementState,
  newPassword: initialElementState,
  newPasswordConfirm: initialElementState,
};

export const useChangePasswordState = (memberUuid: string) => {
  const [formState, setFormState] = useState<CurrentFormState>(initialFormState);

  const { validationResult } = useFormValidation(formState);
  const { mutateAsync, isPending } = useChangePasswordPut(memberUuid);

  const updatePasswordValue = (updatedValue: string, fieldToUpdate: PasswordField) =>
    setFormState((prevState) => ({
      ...prevState,
      [fieldToUpdate]: { value: updatedValue },
    }));

  const updatePasswordError = (errorMessage: string, fieldToUpdate: PasswordField) =>
    setFormState((prevState) => ({
      ...prevState,
      [fieldToUpdate]: { ...prevState[fieldToUpdate], error: errorMessage },
    }));

  const onFieldBlur = (blurredField: PasswordField) => {
    if (!validationResult.success) {
      const fieldError = validationResult.error.issues.find((issue) =>
        issue.path.includes(blurredField),
      );

      updatePasswordError(fieldError?.message ?? '', blurredField);
    }
  };

  const makeApiRequest = async () => {
    return mutateAsync({
      currentPassword: formState.currentPassword.value,
      newPassword: formState.newPassword.value,
    });
  };

  return {
    formState,
    saveDisabled: !validationResult.success || isPending,
    updatePasswordValue,
    updatePasswordError,
    onFieldBlur,
    makeApiRequest,
  };
};
