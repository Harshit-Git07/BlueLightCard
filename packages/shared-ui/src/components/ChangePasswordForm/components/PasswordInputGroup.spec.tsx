import { PasswordInputGroup, PasswordInputGroupProps } from './PasswordInputGroup';
import { initialFormState } from '../useChangePasswordState';
import { act, render, screen } from '@testing-library/react';
import { userEvent } from '@storybook/testing-library';

const mockUpdateValue = jest.fn();
const mockBlur = jest.fn();
describe('PasswordInputGroup', () => {
  const defaultProps: PasswordInputGroupProps = {
    formState: initialFormState,
    updatePasswordValue: mockUpdateValue,
    onBlur: mockBlur,
  };

  it('matches snapshot in default state', () => {
    const { container } = render(<PasswordInputGroup {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });

  it('matches snapshot in populated state', () => {
    const { container } = render(
      <PasswordInputGroup
        {...defaultProps}
        formState={{
          currentPassword: { value: 'testCurrentPassword' },
          newPassword: { value: 'testNewPassword' },
          newPasswordConfirm: { value: 'testNewPasswordConfirm' },
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('triggers callback with correct params when inputting values into correct elements', async () => {
    render(<PasswordInputGroup {...defaultProps} />);

    const currentPasswordInputField = screen.getAllByLabelText('Current password')[1];
    await act(async () => {
      await userEvent.type(currentPasswordInputField, 'testCurrentPassword');
    });
    expect(mockUpdateValue).toHaveBeenLastCalledWith('testCurrentPassword', 'currentPassword');

    const newPasswordInputField = screen.getAllByLabelText('New password')[1];
    await act(async () => {
      await userEvent.type(newPasswordInputField, 'testNewPassword');
    });
    expect(mockUpdateValue).toHaveBeenLastCalledWith('testNewPassword', 'newPassword');

    const newPasswordConfirmInputField = screen.getAllByLabelText('Confirm new password')[1];
    await act(async () => {
      await userEvent.type(newPasswordConfirmInputField, 'testNewPasswordConfirm');
    });
    expect(mockUpdateValue).toHaveBeenLastCalledWith(
      'testNewPasswordConfirm',
      'newPasswordConfirm',
    );
  });

  it('triggers blur callback with correct params when blurring field', async () => {
    render(<PasswordInputGroup {...defaultProps} />);

    const currentPasswordInputField = screen.getAllByLabelText('Current password')[1];
    await act(async () => {
      await userEvent.type(currentPasswordInputField, ' ');
      await userEvent.tab();
    });
    expect(mockBlur).toHaveBeenLastCalledWith('currentPassword');

    const newPasswordInputField = screen.getAllByLabelText('New password')[1];
    await act(async () => {
      await userEvent.type(newPasswordInputField, ' ');
      await userEvent.tab();
    });
    expect(mockBlur).toHaveBeenLastCalledWith('newPassword');

    const newPasswordConfirmInputField = screen.getAllByLabelText('Confirm new password')[1];
    await act(async () => {
      await userEvent.type(newPasswordConfirmInputField, ' ');
      await userEvent.tab();
    });
    expect(mockBlur).toHaveBeenLastCalledWith('newPasswordConfirm');
  });
});
