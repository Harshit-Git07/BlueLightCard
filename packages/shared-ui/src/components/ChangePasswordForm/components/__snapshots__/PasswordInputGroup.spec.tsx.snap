// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInputGroup matches snapshot in default state 1`] = `
<div>
  <div
    class="flex flex-col gap-4"
  >
    <div>
      <div>
        <label
          aria-label="Current password"
          class="mr-[8px] text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
          for="password-:r0:"
        >
          Current password
        </label>
      </div>
      <p
        class="mt-[6px] text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
        id="helpMessage-:r0:"
      />
      <div
        class="relative mt-[6px]"
      >
        <input
          aria-disabled="false"
          aria-required="true"
          class="peer w-full h-[50px] px-[16px] pt-[16px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body text-colour-onSurface dark:text-colour-onSurface-dark bg-transparent rounded-md border outline-none border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
          id="password-:r0:"
          type="password"
        />
        <label
          aria-disabled="false"
          aria-hidden="false"
          class="left-4 -translate-y-1/2 absolute transition-all pointer-events-none top-[25px] peer-focus:top-4 font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body peer-focus:font-typography-body-light peer-focus:font-typography-body-light-weight peer-focus:text-typography-body-light peer-focus:leading-typography-body-light peer-focus:tracking-typography-body-light peer-focus:text-xs text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          for="password-:r0:"
        >
          Add current password
        </label>
        <button
          aria-label="Show password"
          class="absolute top-1/2 right-4 -translate-y-1/2 false"
          title="Show password"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye w-[16px] h-[16px] text-colour-onSurface dark:text-colour-onSurface-dark"
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288 32c-80.8 0-145.5 36.8-192.6 80.6C48.6 156 17.3 208 2.5 243.7c-3.3 7.9-3.3 16.7 0 24.6C17.3 304 48.6 356 95.4 399.4C142.5 443.2 207.2 480 288 480s145.5-36.8 192.6-80.6c46.8-43.5 78.1-95.4 93-131.1c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C433.5 68.8 368.8 32 288 32zM144 256a144 144 0 1 1 288 0 144 144 0 1 1 -288 0zm144-64c0 35.3-28.7 64-64 64c-7.1 0-13.9-1.2-20.3-3.3c-5.5-1.8-11.9 1.6-11.7 7.4c.3 6.9 1.3 13.8 3.2 20.7c13.7 51.2 66.4 81.6 117.6 67.9s81.6-66.4 67.9-117.6c-11.1-41.5-47.8-69.4-88.6-71.1c-5.8-.2-9.2 6.1-7.4 11.7c2.1 6.4 3.3 13.2 3.3 20.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
    <div>
      <div>
        <label
          aria-label="New password"
          class="mr-[8px] text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
          for="password-:r1:"
        >
          New password
        </label>
      </div>
      <p
        class="mt-[6px] text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
        id="helpMessage-:r1:"
      />
      <div
        class="relative mt-[6px]"
      >
        <input
          aria-disabled="false"
          aria-required="true"
          class="peer w-full h-[50px] px-[16px] pt-[16px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body text-colour-onSurface dark:text-colour-onSurface-dark bg-transparent rounded-md border outline-none border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
          id="password-:r1:"
          type="password"
        />
        <label
          aria-disabled="false"
          aria-hidden="false"
          class="left-4 -translate-y-1/2 absolute transition-all pointer-events-none top-[25px] peer-focus:top-4 font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body peer-focus:font-typography-body-light peer-focus:font-typography-body-light-weight peer-focus:text-typography-body-light peer-focus:leading-typography-body-light peer-focus:tracking-typography-body-light peer-focus:text-xs text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          for="password-:r1:"
        >
          Add new password
        </label>
        <button
          aria-label="Show password"
          class="absolute top-1/2 right-4 -translate-y-1/2 false"
          title="Show password"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye w-[16px] h-[16px] text-colour-onSurface dark:text-colour-onSurface-dark"
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288 32c-80.8 0-145.5 36.8-192.6 80.6C48.6 156 17.3 208 2.5 243.7c-3.3 7.9-3.3 16.7 0 24.6C17.3 304 48.6 356 95.4 399.4C142.5 443.2 207.2 480 288 480s145.5-36.8 192.6-80.6c46.8-43.5 78.1-95.4 93-131.1c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C433.5 68.8 368.8 32 288 32zM144 256a144 144 0 1 1 288 0 144 144 0 1 1 -288 0zm144-64c0 35.3-28.7 64-64 64c-7.1 0-13.9-1.2-20.3-3.3c-5.5-1.8-11.9 1.6-11.7 7.4c.3 6.9 1.3 13.8 3.2 20.7c13.7 51.2 66.4 81.6 117.6 67.9s81.6-66.4 67.9-117.6c-11.1-41.5-47.8-69.4-88.6-71.1c-5.8-.2-9.2 6.1-7.4 11.7c2.1 6.4 3.3 13.2 3.3 20.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
    <div>
      <div>
        <label
          aria-label="Confirm new password"
          class="mr-[8px] text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
          for="password-:r2:"
        >
          Confirm new password
        </label>
      </div>
      <p
        class="mt-[6px] text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
        id="helpMessage-:r2:"
      />
      <div
        class="relative mt-[6px]"
      >
        <input
          aria-disabled="false"
          aria-required="true"
          class="peer w-full h-[50px] px-[16px] pt-[16px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body text-colour-onSurface dark:text-colour-onSurface-dark bg-transparent rounded-md border outline-none border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
          id="password-:r2:"
          type="password"
        />
        <label
          aria-disabled="false"
          aria-hidden="false"
          class="left-4 -translate-y-1/2 absolute transition-all pointer-events-none top-[25px] peer-focus:top-4 font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body peer-focus:font-typography-body-light peer-focus:font-typography-body-light-weight peer-focus:text-typography-body-light peer-focus:leading-typography-body-light peer-focus:tracking-typography-body-light peer-focus:text-xs text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          for="password-:r2:"
        >
          Confirm new password
        </label>
        <button
          aria-label="Show password"
          class="absolute top-1/2 right-4 -translate-y-1/2 false"
          title="Show password"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye w-[16px] h-[16px] text-colour-onSurface dark:text-colour-onSurface-dark"
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288 32c-80.8 0-145.5 36.8-192.6 80.6C48.6 156 17.3 208 2.5 243.7c-3.3 7.9-3.3 16.7 0 24.6C17.3 304 48.6 356 95.4 399.4C142.5 443.2 207.2 480 288 480s145.5-36.8 192.6-80.6c46.8-43.5 78.1-95.4 93-131.1c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C433.5 68.8 368.8 32 288 32zM144 256a144 144 0 1 1 288 0 144 144 0 1 1 -288 0zm144-64c0 35.3-28.7 64-64 64c-7.1 0-13.9-1.2-20.3-3.3c-5.5-1.8-11.9 1.6-11.7 7.4c.3 6.9 1.3 13.8 3.2 20.7c13.7 51.2 66.4 81.6 117.6 67.9s81.6-66.4 67.9-117.6c-11.1-41.5-47.8-69.4-88.6-71.1c-5.8-.2-9.2 6.1-7.4 11.7c2.1 6.4 3.3 13.2 3.3 20.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`PasswordInputGroup matches snapshot in populated state 1`] = `
<div>
  <div
    class="flex flex-col gap-4"
  >
    <div>
      <div>
        <label
          aria-label="Current password"
          class="mr-[8px] text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
          for="password-:r3:"
        >
          Current password
        </label>
      </div>
      <p
        class="mt-[6px] text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
        id="helpMessage-:r3:"
      />
      <div
        class="relative mt-[6px]"
      >
        <input
          aria-disabled="false"
          aria-required="true"
          class="peer w-full h-[50px] px-[16px] pt-[16px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body text-colour-onSurface dark:text-colour-onSurface-dark bg-transparent rounded-md border outline-none border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
          id="password-:r3:"
          type="password"
        />
        <label
          aria-disabled="false"
          aria-hidden="false"
          class="left-4 -translate-y-1/2 absolute transition-all pointer-events-none top-4 text-xs font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light tracking-typography-body-light text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          for="password-:r3:"
        >
          Add current password
        </label>
        <button
          aria-label="Show password"
          class="absolute top-1/2 right-4 -translate-y-1/2 false"
          title="Show password"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye w-[16px] h-[16px] text-colour-onSurface dark:text-colour-onSurface-dark"
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288 32c-80.8 0-145.5 36.8-192.6 80.6C48.6 156 17.3 208 2.5 243.7c-3.3 7.9-3.3 16.7 0 24.6C17.3 304 48.6 356 95.4 399.4C142.5 443.2 207.2 480 288 480s145.5-36.8 192.6-80.6c46.8-43.5 78.1-95.4 93-131.1c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C433.5 68.8 368.8 32 288 32zM144 256a144 144 0 1 1 288 0 144 144 0 1 1 -288 0zm144-64c0 35.3-28.7 64-64 64c-7.1 0-13.9-1.2-20.3-3.3c-5.5-1.8-11.9 1.6-11.7 7.4c.3 6.9 1.3 13.8 3.2 20.7c13.7 51.2 66.4 81.6 117.6 67.9s81.6-66.4 67.9-117.6c-11.1-41.5-47.8-69.4-88.6-71.1c-5.8-.2-9.2 6.1-7.4 11.7c2.1 6.4 3.3 13.2 3.3 20.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
    <div>
      <div>
        <label
          aria-label="New password"
          class="mr-[8px] text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
          for="password-:r4:"
        >
          New password
        </label>
      </div>
      <p
        class="mt-[6px] text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
        id="helpMessage-:r4:"
      />
      <div
        class="relative mt-[6px]"
      >
        <input
          aria-disabled="false"
          aria-required="true"
          class="peer w-full h-[50px] px-[16px] pt-[16px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body text-colour-onSurface dark:text-colour-onSurface-dark bg-transparent rounded-md border outline-none border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
          id="password-:r4:"
          type="password"
        />
        <label
          aria-disabled="false"
          aria-hidden="false"
          class="left-4 -translate-y-1/2 absolute transition-all pointer-events-none top-4 text-xs font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light tracking-typography-body-light text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          for="password-:r4:"
        >
          Add new password
        </label>
        <button
          aria-label="Show password"
          class="absolute top-1/2 right-4 -translate-y-1/2 false"
          title="Show password"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye w-[16px] h-[16px] text-colour-onSurface dark:text-colour-onSurface-dark"
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288 32c-80.8 0-145.5 36.8-192.6 80.6C48.6 156 17.3 208 2.5 243.7c-3.3 7.9-3.3 16.7 0 24.6C17.3 304 48.6 356 95.4 399.4C142.5 443.2 207.2 480 288 480s145.5-36.8 192.6-80.6c46.8-43.5 78.1-95.4 93-131.1c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C433.5 68.8 368.8 32 288 32zM144 256a144 144 0 1 1 288 0 144 144 0 1 1 -288 0zm144-64c0 35.3-28.7 64-64 64c-7.1 0-13.9-1.2-20.3-3.3c-5.5-1.8-11.9 1.6-11.7 7.4c.3 6.9 1.3 13.8 3.2 20.7c13.7 51.2 66.4 81.6 117.6 67.9s81.6-66.4 67.9-117.6c-11.1-41.5-47.8-69.4-88.6-71.1c-5.8-.2-9.2 6.1-7.4 11.7c2.1 6.4 3.3 13.2 3.3 20.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
    <div>
      <div>
        <label
          aria-label="Confirm new password"
          class="mr-[8px] text-colour-onSurface dark:text-colour-onSurface-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
          for="password-:r5:"
        >
          Confirm new password
        </label>
      </div>
      <p
        class="mt-[6px] text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body"
        id="helpMessage-:r5:"
      />
      <div
        class="relative mt-[6px]"
      >
        <input
          aria-disabled="false"
          aria-required="true"
          class="peer w-full h-[50px] px-[16px] pt-[16px] font-typography-body font-typography-body-weight text-typography-body leading-typography-body tracking-typography-body text-colour-onSurface dark:text-colour-onSurface-dark bg-transparent rounded-md border outline-none border-colour-onSurface-outline dark:border-colour-onSurface-outline-dark"
          id="password-:r5:"
          type="password"
        />
        <label
          aria-disabled="false"
          aria-hidden="false"
          class="left-4 -translate-y-1/2 absolute transition-all pointer-events-none top-4 text-xs font-typography-body-light font-typography-body-light-weight text-typography-body-light leading-typography-body-light tracking-typography-body-light text-colour-onSurface-subtle dark:text-colour-onSurface-subtle-dark"
          for="password-:r5:"
        >
          Confirm new password
        </label>
        <button
          aria-label="Show password"
          class="absolute top-1/2 right-4 -translate-y-1/2 false"
          title="Show password"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-eye w-[16px] h-[16px] text-colour-onSurface dark:text-colour-onSurface-dark"
            data-icon="eye"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288 32c-80.8 0-145.5 36.8-192.6 80.6C48.6 156 17.3 208 2.5 243.7c-3.3 7.9-3.3 16.7 0 24.6C17.3 304 48.6 356 95.4 399.4C142.5 443.2 207.2 480 288 480s145.5-36.8 192.6-80.6c46.8-43.5 78.1-95.4 93-131.1c3.3-7.9 3.3-16.7 0-24.6c-14.9-35.7-46.2-87.7-93-131.1C433.5 68.8 368.8 32 288 32zM144 256a144 144 0 1 1 288 0 144 144 0 1 1 -288 0zm144-64c0 35.3-28.7 64-64 64c-7.1 0-13.9-1.2-20.3-3.3c-5.5-1.8-11.9 1.6-11.7 7.4c.3 6.9 1.3 13.8 3.2 20.7c13.7 51.2 66.4 81.6 117.6 67.9s81.6-66.4 67.9-117.6c-11.1-41.5-47.8-69.4-88.6-71.1c-5.8-.2-9.2 6.1-7.4 11.7c2.1 6.4 3.3 13.2 3.3 20.3z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
